src/student_registry_project/ex_1.rs

```diff
@@ -0,0 +1,384 @@
+// use student_registry_project::{
+use crate::student_registry_project::types::basic_types::{Sex,Course, Student, StudentRegistry};
+
+// use types::basic_types::{Sex, Student, StudentRegistry};
+// use utils::convert_to_string;
+// };
+
+use std::io;
+use std::io::Write; // for flush
+
+// fn create_course(name: String, capacity: u32, mut _courses: Vec<Course>) -> Vec<Course> {
+//     let course_id: u32 = _courses.len() as u32 + 1;
+//     let new_course = Course {
+//         id: course_id,
+//         name: name,
+//         capacity: capacity,
+//         enrolled_student_ids: Vec::new(),
+//     };
+//     _courses.push(new_course);
+//     _courses
+// }
+
+// fn prompt(input: &str) {
+//     print!("{input}");
+//     io::stdout().flush().unwrap();
+    
+// }
+fn get_courses(rust_cohort: &mut StudentRegistry) -> u32 {
+    print!("{:#?}", rust_cohort.course_registry);
+    io::stdout().flush().unwrap();
+
+    print!("Enter course id: \n");
+    io::stdout().flush().unwrap();
+    let mut course_id = String::new();
+    io::stdin()
+        .read_line(&mut course_id)
+        .expect("Failed to read line");
+    let course_id = course_id
+        .trim()
+        .parse::<u32>()
+        .expect("Invalid input. Please enter a number.");
+    // let course = rust_cohort.get_course_by_id(course_id);
+    // println!("course: {course:#?}");
+    course_id
+}
+
+fn student_flow(rust_cohort: &mut StudentRegistry) {
+    let mut student_state: i32 = 0;
+    loop {
+        let mut input = String::new();
+
+        print!("1. Add student\n2. Get student\n3.Enroll student in course\n4.Return\n");
+        io::stdout().flush().unwrap();
+
```

Comment:
Consider implementing utility functions to parse user inputs. This will help keep the code modular and DRY.